### PR TITLE
PAPP-26547: Initial commit

### PR DESCRIPTION
**Pull request by pdesai-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by zamastyle** [comment]
@pdesai-crest 
I don't understand why the latest changes were added. You have added a render dictionary to an action that should not have one and added a bunch of wheels to the app. Can you explain what the purpose of these changes are? This app has been tested on several versions of the platform including the latest and none of these wheels were required for it to run properly.

**Comment by pdesai-crest** [comment]
static tests are failing due to min phantom version is kept at 4.9.0 as suggested by @zamastyle

File: exodus.json [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    "license": "Copyright (c) Mhike, 2022",
+    "python_version": "3",
+    "contributors": [
+        {
+            "name": "Mhike"
+        }
+    ],
+    "configuration": {
+        "source_base_url": {
+            "description": "URL for source Phantom host",
```
Comment (alexa-phantom) [inline exodus.json]: SOAR instead of Phantom

File: exodus.json [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            "data_type": "string",
+            "required": true,
+            "default": "https://devphantom.company.com/",
```
Comment (alexa-phantom) [inline exodus.json]: devsoar -> align with SOAR product name

File: exodus.json (hunk 1 above)
Comment (alexa-phantom) [reply exodus.json]: Please update all other Phantom occurrences to SOAR

File: exodus.json [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            "output": [],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "add approval",
+            "description": "Add approval artifact to container",
+            "verbose": "Add a specifically formatted artifact that is required for the approval process for exodus containers.",
+            "type": "generic",
+            "identifier": "add_approval",
+            "read_only": true,
```
Comment (alexa-phantom) [inline exodus.json]: false

File: exodus.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            "parameters": {},
+            "output": [
```
Comment (alexa-phantom) [inline exodus.json:117]: Possibly return data to the user?

**Review by alexa-phantom — CHANGES_REQUESTED** [review]
There are some pretty cool actions here, but I what are your thoughts on making some fields, such as `label` and `name` custom parameters? ie finding containers under another label aside from `exodus`?

File: exodus.json (hunk 4 above)
Comment (zamastyle) [reply exodus.json:117]: The sole purpose of this action is to add a specific approval artifact to the container. The artifact in itself has been enough so far. The most we could meaningfully return is the success or failure of adding the artifact and the action status essentially does that job. What did you have in mind to return?

File: exodus.json (hunk 3 above)
Comment (zamastyle) [reply exodus.json]: Good fix

File: exodus.json (hunk 2 above)
Comment (zamastyle) [reply exodus.json]: good fix

File: exodus.json (hunk 1 above)
Comment (zamastyle) [reply exodus.json]: good fix

**Comment by zamastyle** [comment]
> There are some pretty cool actions here, but I what are your thoughts on making some fields, such as `label` and `name` custom parameters? ie finding containers under another label aside from `exodus`?

The label config on a per asset basis is solid idea. I haven't seen a good use for it yet but I imagine its out there. The overall idea of the exodus label is to have a limited access container so that only authorized personnel can approve migration. That access control is left to the admin to manage. The single exodus label was just my way of injecting a requirement to add a label and consider the access limits to it. Beyond that there isn't really a good reason to limit the label an exodus asset uses.

As far as name goes, between the container creation and the container approval or rejection, the user can automate however they would like. My general expectation is that the name of the container and other specific aspects of the container would be automatically updated by an active playbook to prep it for whatever tailored review process the admin wants done.  Might be more valuable to create a dynamic name based on the name and id of the object to be migrated instead of adding an option for a different static name.

File: exodus.json (hunk 1 above)
Comment (pdesai-crest) [reply exodus.json]: Changes done

File: exodus.json (hunk 2 above)
Comment (pdesai-crest) [reply exodus.json]: done

File: exodus.json (hunk 3 above)
Comment (pdesai-crest) [reply exodus.json]: done

File: exodus.json (hunk 4 above)
Comment (alexa-phantom) [reply exodus.json:117]: Not necessarily, just wanted to point out that there weren't any fields being outputted to the output table.

**Review by alexa-phantom — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**